ir_tx/nec.py

```diff
@@ -27,7 +27,10 @@ def tx(self, addr, data, _):  # Ignore toggle
         else:
             self.append(9000, 4500)
         if addr < 256:  # Short address: append complement
-            addr |= ((addr ^ 0xff) << 8)
+            if self.samsung:
+              addr |= addr << 8
```

Comment:
This will be incorrect for my Samsung remote. It sounds as if they have more than one standard.

Reply:
There are definitively multiple protocols named "SamsungXX", including _[SAMSUNG32](https://www.mikrocontroller.net/articles/IRMP_-_english#SAMSUNG32)_ and _[SAMSUNG48](https://www.mikrocontroller.net/articles/IRMP_-_english#SAMSUNG48)_

I believe that the work done by the [Flipper Zero team](https://github.com/flipperdevices) which is selling a [product](https://flipperzero.one/) around [their code](https://github.com/flipperdevices/flipperzero-firmware/tree/dev/lib/infrared/encoder_decoder/samsung) is the most correct/functional implementation of the _SAMSUNG32_ protocol that I have found.